e2e/wdio/core/tests/illustrated-message.e2e-spec.ts

```diff
@@ -0,0 +1,108 @@
+            }
+        });
+
+        it('should check back button focus state', () => {
+            const buttonsArrLength = getElementArrayLength(sceneAndSpotButtons);
+            for (let i = 0; buttonsArrLength > i; i++) {
+                checkButtonFocusState(sceneAndSpotButtons, 'FocusState'+ '-' + i, sceneAndSpotButtons, i);
+            }
+        });
+
+        it('should check back button active state', () => {
+            const buttonsArrLength = getElementArrayLength(sceneAndSpotButtons);
+            for (let i = 0; buttonsArrLength > i; i++) {
+                checkButtonActiveState(sceneAndSpotButtons, 'ActiveState'+ '-' + i, sceneAndSpotButtons, i);
+            }
+        });
+    });
+
+    it('should check RTL and LTR orientation', () => {              
+        illustratedMessagePage.checkRtlSwitch();
+    });
+
+    it('should check visual regression for all examples', () => {
+        illustratedMessagePage.saveExampleBaselineScreenshot();
+        expect(illustratedMessagePage.compareWithBaseline()).toBeLessThan(1);
+    });
+
+    it('should check is button clickable', () => {
+        const buttonsArrLength = getElementArrayLength(sceneAndSpotButtons);
+        for (let i = 0; buttonsArrLength > i; i++) {
+            expect(isElementClickable(sceneAndSpotButtons, i)).toBe(true, `Button ${i} is not clickable`);
+        }
+    });
+
+    describe('check dialog example', function() {
+        it('should open dialog popup illustrated message', () => {
+            click(buttonDialog);
+            waitForElDisplayed(dialogPopup);
+            expect(elementDisplayed(dialogPopup)).toBeTruthy();
+        });
+
+        it('should check visual regression dialog popup illustrated message', () => {
+            expect(checkElementScreenshot(dialogPopup, 'dialogPopup', illustratedMessagePage.getScreenshotFolder()))
+                .toBeLessThan(1);
+        }); 
+
+        it('should close dialog popup illustrated message by click on "Close sign X" button', () => {
+            click(closePopupSignButton);
+            expect(doesItExist(dialogPopup)).toBeFalsy();
+        });
+
+        it('should close dialog popup illustrated message by click on "Close" button', () => {
+            click(buttonDialog);
+            click(closePopupButton);
+            expect(doesItExist(dialogPopup)).toBeFalsy();
+        });
+    });
+
+    function checkButtonHoverState(selector: string, tag: string, btnName: string, index: number = 0): void {
+        mouseHoverElement(selector, index);
+        saveElementScreenshot(selector, tag, illustratedMessagePage.getScreenshotFolder(), index);
+        expect(checkElementScreenshot(selector, tag, illustratedMessagePage.getScreenshotFolder(), index))
+            .toBeLessThan(2, `${btnName} button item ${index} hover state mismatch`);
+    }
+
+    function checkButtonFocusState(selector: string, tag: string, btnName: string, index: number = 0): void {
+        click(selector, index);
```

Comment:
better to use focusElement(selector, index) instead of click to make sure that you get the true focus state, rather than focus with active or hover as well.